src/runtime/patch/child_process.js

```diff
@@ -0,0 +1,203 @@
+    if (!!cwdStat && cwdStat.isDirectory() && !existsOnRealFs(cwd)) {
+      realFs.mkdirSync(cwd, { recursive: true });
+    }
+  } catch {
+    /* no-op */
+  }
+}
+
+// Build the args order for the child_process functions
+// The order is always the same:
+// 1 - first add the arguments starting with -- or -
+// 2 - add the main static fs runtime path
+// 3 - add the static-fs-volumes flag
+// 4 - add the main entry
+// 5 - add the rest of the args
+//
+// However for fork the order is different as fork don't accept
+// node args starting by -- or - unless we set the options.execArgs
+// So for fork (when mainStaticFsRuntimePath === null) the order is
+//
+// 1 - add the static-fs-volumes flag
+// 2 - add the main entry
+// 3 - add the rest of the args
+function buildStaticFsArgs(args, mainStaticFsRuntimePath, staticFsVolumesPaths, mainEntry = null) {
+  const sanitizedArgs = Array.isArray(args) ? args : typeof args === 'string' ? args.split(' ') : [];
+  const builtArgs = [];
+  let toAddMetaArgs = true;
+
+  const iterableArgs = [...sanitizedArgs];
+
+  do {
+    const wasArrayEmptyBeforeArg = iterableArgs.length === 0;
+    const arg = iterableArgs.shift();
+
+    if (!arg && !wasArrayEmptyBeforeArg) {
+      return;
+    }
+
+    if ((typeof arg === 'string' && arg.startsWith('-') && mainStaticFsRuntimePath) || !toAddMetaArgs) {
+      builtArgs.push(arg);
+      continue;
+    }
+
+    toAddMetaArgs = false;
+
+    if (mainStaticFsRuntimePath) {
+      builtArgs.push(mainStaticFsRuntimePath);
+    }
+
+    builtArgs.push(`--static-fs-volumes=${staticFsVolumesPaths.join(',')}`);
+
+    if (mainEntry) {
+      builtArgs.push(mainEntry);
+    }
+
+    builtArgs.push(arg);
+  } while (iterableArgs.length > 0);
+
+  if (!builtArgs.length) {
+    throw new Error('Something went wrong building the static fs args for the child_process functions');
+  }
+
+  return builtArgs;
+}
+
+function indexAfterStartsWith(command, text) {
+  return command.startsWith(text) ? text.length : -1;
+}
+
+function isNode(path) {
+  return possibilities.indexOf(path) > -1;
+}
+
+function startsWithNode(command) {
+  if (command.charAt(0) === '"') {
+    // check includes quotes
+    for (const each of possibilities) {
+      const val = indexAfterStartsWith(command, `"${each}" `);
+      if (val > -1) {
+        return val;
+      }
+    }
+  } else {
+    for (const each of possibilities) {
+      const val = indexAfterStartsWith(command, `${each} `);
+      if (val > -1) {
+        return val;
+      }
+    }
+  }
+  return -1;
+}
+
+export function patchChildProcess(sfsRuntime) {
+  const sfs = sfsRuntime.staticfilesystem;
+  const fork = child_process.fork;
+  const spawn = child_process.spawn;
+  const exec = child_process.exec;
+  const spawnSync = child_process.spawnSync;
+  const execSync = child_process.execSync;
+
+  // hot-patch fork so we can make child processes work too.
+  child_process.fork = (modulePath, args, options) => {
```

Comment:
looping in @legrego for this

Reply:
Hope my explanation in the walkthrough for this has been helpful!